charmcraft/commands/build.py

```diff
@@ -218,6 +218,7 @@ def build_charm(self, bases_config: BasesConfiguration) -> str:
         lifecycle = PartsLifecycle(
             self._parts,
             work_dir=self.buildpath,
+            ignore_local_sources=["*.charm"],
```

Comment:
Perhaps I'm misunderstanding this line, but should there be consideration for anything in the `.jujuignore` file too?

Reply:
this should be handled in charm_builder.py (https://github.com/canonical/charmcraft/blob/9dc20872f1715585dfa23e2332bfece47126072f/charmcraft/charm_builder.py#L156) which is prior to this.

Reply:
and that comes from

https://github.com/canonical/charmcraft/blob/9dc20872f1715585dfa23e2332bfece47126072f/charmcraft/charm_builder.py#L78

Reply:
Ah right, could very well be sorted already

Reply:
`.jujuignore` is handled by the charm build logic called from the parts lifecycle. The main purpose of this `*.charm` ignore filter is to prevent the lifecycle to consider the newly created charm file as part of the source, otherwise every time the charm file is created the lifecycle manager will consider the source outdated (because there's a new file there).